src/Sentry/Internal/SerializableExtensions.cs

```diff
@@ -22,15 +22,21 @@ await using (stream.ConfigureAwait(false))
 #endif
             {
                 await serializable.SerializeAsync(stream, logger, cancellationToken).ConfigureAwait(false);
-                return Encoding.UTF8.GetString(stream.ToArray());
+
+                stream.Seek(0, SeekOrigin.Begin);
+                using var reader = new StreamReader(stream);
+                return await reader.ReadToEndAsync().ConfigureAwait(false);
             }
         }
 
         public static string SerializeToString(this ISerializable serializable, IDiagnosticLogger logger)
         {
             using var stream = new MemoryStream();
             serializable.Serialize(stream, logger);
-            return Encoding.UTF8.GetString(stream.ToArray());
```

Comment:
Just for clarity - this is the only real useful change.  We shouldn't call `MemoryStream.ToArray` because it creates an intermediate byte array.  So there were three copies of the data - one in the stream, one in the byte array, and one in the output string.  We only need two.